app/controllers/api/v4/imports_controller.rb

```diff
@@ -18,4 +20,11 @@ def import
   def import_params
     params.require(:resources)
   end
+
+  def validate_token_organization
+    token_organization = MachineUser.find_by(id: doorkeeper_token.application&.owner_id)&.organization_id
+    unless token_organization.present? && token_organization == request.headers["HTTP_X_ORGANIZATION_ID"]
+      head :forbidden
+    end
+  end
```

Comment:
This validates if the Auth token is associated with the organization that the requester intended. Mostly included for defense-in-depth.